src/Cookie.php

```diff
@@ -115,7 +115,7 @@ public function setPath(string $path): void
         $this->path = $path;
     }
 
-    public function getDomain(): string
+    public function getDomain(): ?string
```

Comment:
Not sure if that should be but: Your normalize method return null value what ends up in the case that `$this->domain` is `null`. In that case `getDomain` should return null.